modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/MuleExtensionUtils.java

```diff
@@ -584,4 +601,82 @@ public static boolean isConnectedStreamingOperation(ComponentModel componentMode
+                                                                       "Parameter group '%s' requires that one of its optional parameters should be set but all of them are missing. "
+                                                                           + "One of the following should be set: [%s]",
+                                                                       group.getName(),
+                                                                       Joiner.on(", ")
+                                                                           .join(exclusiveModel
+                                                                               .getExclusiveParameterNames())))));
+        } else if (definedExclusiveParameters.size() > 1) {
+          if (model.isPresent()) {
+            throw new ConfigurationException(createStaticMessage(format("In %s '%s', the following parameters cannot be set at the same time: [%s]",
+                                                                        getComponentModelTypeName(model.get()),
+                                                                        getModelName(model.get()),
+                                                                        Joiner.on(", ").join(definedExclusiveParameters))));
+          } else {
+            throw new ConfigurationException(createStaticMessage(format("The following parameters cannot be set at the same time: [%s]",
+                                                                        Joiner.on(", ").join(definedExclusiveParameters))));
+          }
+        }
+      }
+    }
+  }
+
+  private static Set<String> resolveParameterNames(ParameterGroupModel group, Map<String, ?> parameters,
+                                                   Map<String, String> aliasedParameterNames)
+      throws ConfigurationException {
+    Set<String> topLevelParameterNames = new HashSet<>();
+    parameters.forEach((key, value) -> {
+      aliasedParameterNames.getOrDefault(key, key);
+      topLevelParameterNames.add(key);
+    });
+
+    if (group.isShowInDsl()) {
+      ExtensionParameterDescriptorModelProperty property = group.getModelProperty(ExtensionParameterDescriptorModelProperty.class)
+          .orElseThrow(() -> new ConfigurationException(createStaticMessage("Could not find ExtensionParameterDescriptorModelProperty for the parameter group %s",
+                                                                            group.getName())));
+      String containerName = property.getExtensionParameter().getName();
+      try {
+        Object parameter = parameters.get(containerName);
+        if (parameter == null) {
+          throw new ConfigurationException(createStaticMessage("Was expecting a parameter with name [%s] among the resolved parameters",
+                                                               containerName));
+        }
+        return ((ParameterValueResolver) parameters.get(containerName)).getParameters().keySet().stream()
```

Comment:
I would make a check before casting this, so that we get a better error message of this is not a ParameterValueResolver.